app/gui/view/src/project.rs

```diff
@@ -263,12 +267,26 @@ impl Model {
         }
     }
 
-    fn on_dom_shape_changed(&self, shape: &dom::shape::Shape) {
-        // Top buttons must always stay in top-left corner.
-        if let Some(window_control_buttons) = &*self.window_control_buttons {
-            let pos = Vector2(-shape.width, shape.height) / 2.0;
-            window_control_buttons.set_xy(pos);
-        }
+    // Relayouts the top bar elements and returns the new width of the top bar part of the project
+    // view.
+    fn relayout_top_bar(
+        &self,
+        scene_shape: &display::scene::Shape,
+        window_control_buttons_width: f32,
+    ) -> f32 {
+        let top_left = Vector2(-scene_shape.width, scene_shape.height) / 2.0;
+        self.window_control_buttons.set_xy(top_left);
+        let go_to_dashboard_button_offset = Vector2(window_control_buttons_width, 0.0);
+        let go_to_dashboard_button_origin = Vector2(
+            go_to_dashboard_button::SIZE / 2.0,
+            crate::graph_editor::MACOS_TRAFFIC_LIGHTS_VERTICAL_CENTER,
+        );
+        let go_to_dashboard_button_pos =
+            top_left + go_to_dashboard_button_offset + go_to_dashboard_button_origin;
+        self.go_to_dashboard_button.set_xy(go_to_dashboard_button_pos);
+        let project_view_top_bar_width =
+            go_to_dashboard_button_offset.x + go_to_dashboard_button::SIZE;
+        project_view_top_bar_width
```

Comment:
I was writing about in in my previous PR review - this can use auto-layout for window control buttons and the "go to dashboard" button. Is there a reason auto-layout is not used here?

Reply:
I have a reason: it wasn't there before, and was implemented this way. Changing to automatic layouts could be smooth, but also it could not.

I'd wait with changing to autolayout to a moment when we make a new, single top bar component.